compute_endpoint/globus_compute_endpoint/engines/base.py

```diff
@@ -117,33 +117,47 @@ def _status_report(
             packed = messagepack.pack(status_report)
             self.results_passthrough.put(packed)
 
-    def _future_done_callback(self, future: Future):
-        """Callback to post result to the passthrough queue
+    def _setup_future_done_callback(self, task_id: uuid.UUID, future: Future) -> None:
+        """
+        Set up the done() callback for the provided future.
+
+        The done callback handles
+        Callback to post result to the passthrough queue
         Parameters
         ----------
         future: Future for which the callback is triggerd
         """
 
-        if future.exception():
-            code, user_message = get_result_error_details()
-            error_details = {"code": code, "user_message": user_message}
-            exec_end = TaskTransition(
-                timestamp=time.time_ns(),
-                state=TaskState.EXEC_END,
-                actor=ActorName.WORKER,
-            )
-            result_message = dict(
-                task_id=future.task_id,  # type: ignore
-                data=get_error_string(),
-                exception=get_error_string(),
-                error_details=error_details,
-                task_statuses=[exec_end],  # We don't have any more info transitions
-            )
-            packed_result = messagepack.pack(Result(**result_message))
-        else:
-            packed_result = future.result()
-
-        self.results_passthrough.put(packed_result)
+        exec_beg = TaskTransition(
+            timestamp=time.time_ns(),
+            actor=ActorName.INTERCHANGE,
+            state=TaskState.WAITING_FOR_LAUNCH,
+        )
+
+        def _done_cb(f: Future):
+            if f.exception():
+                exc = f.exception()
+                code, user_message = get_result_error_details(exc)
```

Comment:
Good catch here! This was probably bugging out reporting executor errors for a while